### DISCUSSION TITLE
Example of CLOG Database and some talk about future

### DISCUSSION THREAD

**Discussion by rabbibotton** [rabbibotton/clog#115] [Show and tell]
I am working on getting out the full suite of database controls. So far CLOG Database and CLOG One Row are in the builder. Between the two already easy to produce  basic database forms. If you would like to take a look there is a test file rolodex.clog in the test directory. It uses an in memory sqlight3 database can modify that to a file if wish.

As these are the first set of "custom" controls beyond the base as I build them I am building up the needed hooks etc to allow creating custom components by users for the future. 

Also laying the foundation for the more advanced features of the builder that will include pre-canned actions for events.

**Comment by rabbibotton** [comment]
I have expanded rolodex.clog to show how one-to-one or one-to-many relations are setup between clog-one-row controls. This automates easily the relationships.

**Comment by rabbibotton** [comment]
I just added CLOG-Lookup that in the builder is user for Table Lookups for list boxes and drop down boxes. The control is dropped on to a database, a clog-one-row or another clog-lookup (for master slave relationships) and walla super amazing relation database panels :) test/drop.clog contains an text. I am almost ready for a full database demo. I want to create a clog-sheet that expands on the list box to a full spreadsheet like (including editing) interface, hopefully today or tomorrow done. I am so excited :)

**Reply by aykaramba** [reply]
Can't wait to see it.  This is amazing.

**Comment by rabbibotton** [comment]
We are part way to clog-sheet with clog-db-table (technically you can already use clog-db-table but for input requires you to program more than I like) and I may delay a bit on that and also clog-db-panels (a way to express panels on a per row basis) till I do some more testing on creating composite controls from panels.

So as it stands at this point we have the following new database controls:
1) clog-database - connection to a database (uses dbi so sqlite, mysql, postgresql and oracle)
2) clog-one-row - row by row access to a table/query, auto populates controls on panel
3) clog-db-table - html tables generated from database tables and queries
4) clog-lookup - (which is both a listbox and a dropdown control in builder)

All controls can be master and slave between each other to easily create complex database relations outside of sql and using sql together.

Going to try and create this week a nice example using all the control types interacting etc.

**Comment by rabbibotton** [comment]
I am working on creating an app called CLOS-CONTACT with a complete GUI, 3 panels, some report etc. If any one wants can follow along with the progress https://docs.google.com/document/d/185Sh9rMhSCEwt14chbn4_gi0CV6xwWs8Z59rUvEWCN8/edit?usp=sharing

I hope to be done in a day or two.

**Reply by rabbibotton** [reply]
The doc is complete for now. I promoted it on reedit. Going to clean up some things and then publish work to date as CLOG 1.2 and focus 1.3 on polish and writing and publishing extensions.

**Reply by aykaramba** [reply]
Thanks!  Checking it now.
